### PR TITLE
Process wires overriding

### PR DESCRIPTION
**Pull request by Thenerdstation**
**Context:**
Increase speed of wire subset method.

**Description of the Change:**
Previously, the operation `Wires.subset` was around 20% of the total tape construction time. That has been cut down to just 9% by bypassing the `_process` method.

Benchmarks from my laptop.

```
$ asv compare master process_wires

All benchmarks:

       before           after         ratio
     [599e3186]       [360b8357]
     <master>         <process_wires>
       6.06±0.1ms      5.58±0.08ms     0.92  asv.core_suite.CircuitEvaluation.time_circuit(10, 3)
       11.3±0.2ms       10.3±0.4ms    ~0.91  asv.core_suite.CircuitEvaluation.time_circuit(10, 6)
-      16.4±0.4ms       14.3±0.1ms     0.87  asv.core_suite.CircuitEvaluation.time_circuit(10, 9)
      1.88±0.07ms      1.80±0.05ms     0.96  asv.core_suite.CircuitEvaluation.time_circuit(2, 3)
      2.61±0.05ms      2.59±0.06ms     1.00  asv.core_suite.CircuitEvaluation.time_circuit(2, 6)
      3.52±0.06ms      3.31±0.09ms     0.94  asv.core_suite.CircuitEvaluation.time_circuit(2, 9)
      3.19±0.07ms      2.91±0.03ms     0.91  asv.core_suite.CircuitEvaluation.time_circuit(5, 3)
      5.36±0.08ms      4.97±0.04ms     0.93  asv.core_suite.CircuitEvaluation.time_circuit(5, 6)
       7.25±0.1ms       6.69±0.1ms     0.92  asv.core_suite.CircuitEvaluation.time_circuit(5, 9)
       7.94±0.2ms       8.02±0.3ms     1.01  asv.core_suite.GradientComputation.time_gradient_autograd(2, 3)
      22.2±0.08ms       21.8±0.6ms     0.98  asv.core_suite.GradientComputation.time_gradient_autograd(2, 6)
       41.4±0.6ms         40.9±1ms     0.99  asv.core_suite.GradientComputation.time_gradient_autograd(5, 3)
          146±4ms          144±3ms     0.98  asv.core_suite.GradientComputation.time_gradient_autograd(5, 6)
       11.4±0.3ms       11.2±0.1ms     0.98  asv.core_suite.GradientComputation.time_gradient_tf(2, 3)
       27.9±0.3ms       27.9±0.5ms     1.00  asv.core_suite.GradientComputation.time_gradient_tf(2, 6)
       46.9±0.5ms         47.2±1ms     1.01  asv.core_suite.GradientComputation.time_gradient_tf(5, 3)
          155±3ms          155±1ms     1.00  asv.core_suite.GradientComputation.time_gradient_tf(5, 6)
      8.23±0.09ms       8.00±0.1ms     0.97  asv.core_suite.GradientComputation.time_gradient_torch(2, 3)
       21.5±0.5ms       21.8±0.2ms     1.01  asv.core_suite.GradientComputation.time_gradient_torch(2, 6)
         40.8±1ms       41.1±0.3ms     1.01  asv.core_suite.GradientComputation.time_gradient_torch(5, 3)
          147±2ms          144±1ms     0.98  asv.core_suite.GradientComputation.time_gradient_torch(5, 6)

```
**Benefits:**
Wire construction is faster

**Possible Drawbacks:**
None

**Related GitHub Issues:**
#967

### CODE REVIEW

**Comment by Thenerdstation** [comment]
Codecov is being mean again lol

**Comment by josh146** [comment]
[Looking at the CI logs](https://github.com/PennyLaneAI/pennylane/pull/1010/checks?check_run_id=1724370819#step:12:26):

```python
 ==> GitHub Actions detected.
->  Issue detecting commit SHA. Please run actions/checkout with fetch-depth > 1 or set to 0
```

🤔 So it looks like codecov is using the wrong commit to generate the `codecov/patch` check. Note, though, that this check is not a required check!

**Comment by Thenerdstation** [comment]
>Please run actions/checkout with fetch-depth > 1 or set to 0

Is this something we can fix?

**Comment by josh146** [comment]
You could try editing the `.github/workflows/tests.yml` action to have the following:

```python
  with:
    fetch-depth: 0
```

I'm not sure what this does, but the log tells us to do it 😆 

Oh there are more details here: https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches

File: pennylane/wires.py [hunk 1]
```diff
@@ -76,8 +76,11 @@ class Wires(Sequence):
             and hence interpreted as a single wire.
     """
 
-    def __init__(self, wires):
-        self._labels = _process(wires)
+    def __init__(self, wires, _override=False):
```
Comment (mariaschuld) [inline pennylane/wires.py:79]: I have never seen "private" kwargs...is there a convention about this?

File: pennylane/wires.py [hunk 2]
```diff
@@ -303,8 +306,8 @@ def subset(self, indices, periodic_boundary=False):
                     "Cannot subset wire at index {} from {} wires.".format(i, len(self._labels))
                 )
 
-        subset = [self[i] for i in indices]
-        return Wires(subset)
+        subset = tuple(self._labels[i] for i in indices)
+        return Wires(subset, _override=True)
```
Comment (mariaschuld) [inline pennylane/wires.py:310]: Nice!

File: tests/test_wires.py [hunk 3]
```diff
@@ -348,3 +348,8 @@ def test_unique_wires_method(self):
 
         with pytest.raises(WireError, match="Expected a Wires object"):
             Wires.unique_wires([[2, 1], [8, 5]])
+
+    def test_equal_to_tuple(self):
+        assert Wires([1, 2, 3]) == (1, 2, 3)
+        assert Wires([1, 2, 3]) != (1, 5, 3)
+        assert (1, 5, 3) != Wires([1, 2, 3])
```
Comment (mariaschuld) [inline tests/test_wires.py:355]: Also nice!

**Review by mariaschuld — APPROVED** [review]
Awesome: small, safe changes that lead to speedups :)

**Review by mariaschuld — APPROVED** [review]

File: pennylane/wires.py (hunk 1 above)
Comment (Thenerdstation) [reply pennylane/wires.py:79]: It's to discourage users from using it.

**Pull request merged**